lib/base-service.ts

```diff
@@ -205,15 +214,22 @@ export class BaseService {
     const properties = readExternalSources(serviceName);
 
     if (properties !== null) {
-      // the user can define two client-level variables in the credentials file: url and disableSsl
-      const { url, disableSsl } = properties;
+      // the user can define the following client-level variables in the credentials file:
+      // - url
+      // - disableSsl
+      // - enableGzip
+
+      const { url, disableSsl, enableGzip } = properties;
 
       if (url) {
         results.serviceUrl = stripTrailingSlash(url);
       }
       if (disableSsl === true) {
         results.disableSslVerification = disableSsl;
       }
+      if (enableGzip === true) {
```

Comment:
If the `<service>_ENABLE_GZIP` property is defined with any value at all, then we want to use the value whether it is "true" or "false".    This behavior will allow for a default value being set in the service instance by the generated service ctor, and will allow the user to override that default value (in either direction) by setting the `<service>_ENABLE_GZIP` config property to the opposite value.

Reply:
The way I'm handling this is by storing a Boolean value in the request wrapper instance stored in the base service, because that's the class that actually uses the value. When the instance is created, the stored variable is [set to `true` if explicitly set and `false` if not](https://github.com/IBM/node-sdk-core/pull/111/files#diff-3cf5813ed1f8d97bdaad8c743e775fd8R40). When the user overrides the variable with the setter, the value in the request wrapper is updated.

I think this behavior exactly matches what you are describing, I'm just keeping the value buried in the request wrapper (where it will be used) rather than storing it in two different places.